packages/ember/tests/helpers/link_to_test/link_to_with_query_params_test.js

```diff
@@ -1,506 +1,567 @@
+      {{link-to 'Baz Foos' 'foos' (query-params baz=true) id='baz-foos-link'}}
+      {{link-to 'Quux Bars' 'bars' (query-params quux=true) id='bars-link'}}
+    `);
+    this.add('controller:foos', Controller.extend({
+      queryParams: ['status'],
+      baz: false
+    }));
+    this.add('route:foos', Route.extend({
+      model() {
+        return foos.promise;
+      }
+    }));
+    this.add('controller:bars', Controller.extend({
+      queryParams: ['status'],
+      quux: false
+    }));
+    this.add('route:bars', Route.extend({
+      model() {
+        return bars.promise;
+      }
+    }));
 
-QUnit.test('link-to default query params while in active transition regression test', function() {
-  App.Router.map(function() {
-    this.route('foos');
-    this.route('bars');
-  });
-  let foos = RSVP.defer();
-  let bars = RSVP.defer();
-
-  setTemplate('application', compile(`
-    {{link-to 'Foos' 'foos' id='foos-link'}}
-    {{link-to 'Baz Foos' 'foos' (query-params baz=true) id='baz-foos-link'}}
-    {{link-to 'Quux Bars' 'bars' (query-params quux=true) id='bars-link'}}
-  `));
-
-  App.FoosController = Controller.extend({
-    queryParams: ['status'],
-    baz: false
-  });
-
-  App.FoosRoute = Route.extend({
-    model() {
-      return foos.promise;
-    }
-  });
+    this.visit('/').then(() => {
+      let router = this.appRouter;
+      let foosLink = this.$('#foos-link');
+      let barsLink = this.$('#bars-link');
+      let bazLink = this.$('#baz-foos-link');
 
-  App.BarsController = Controller.extend({
-    queryParams: ['status'],
-    quux: false
-  });
+      equal(foosLink.attr('href'), '/foos');
+      equal(bazLink.attr('href'), '/foos?baz=true');
+      equal(barsLink.attr('href'), '/bars?quux=true');
+      equal(router.get('location.path'), '/');
+      this.shouldNotBeActive('#foos-link');
+      this.shouldNotBeActive('#baz-foos-link');
+      this.shouldNotBeActive('#bars-link');
 
-  App.BarsRoute = Route.extend({
-    model() {
-      return bars.promise;
-    }
-  });
+      run(() => barsLink.click());
+      this.shouldNotBeActive('#bars-link');
 
-  bootApplication();
-  equal(jQuery('#foos-link').attr('href'), '/foos');
-  equal(jQuery('#baz-foos-link').attr('href'), '/foos?baz=true');
-  equal(jQuery('#bars-link').attr('href'), '/bars?quux=true');
+      run(() => foosLink.click());
+      this.shouldNotBeActive('#foos-link');
 
-  equal(router.get('location.path'), '');
+      run(foos, 'resolve');
 
-  shouldNotBeActive('#foos-link');
-  shouldNotBeActive('#baz-foos-link');
-  shouldNotBeActive('#bars-link');
+      equal(router.get('location.path'), '/foos');
+      this.shouldBeActive('#foos-link');
+    });
+  }
+});
 
-  run(jQuery('#bars-link'), 'click');
-  shouldNotBeActive('#bars-link');
+moduleFor('The {{link-to}} helper + query params - globals mode app', class extends AutobootApplicationTestCase {
+  [`@test the {{link-to}} helper throws a useful error if you invoke it wrong`](assert) {
```

Comment:
Please add a link back to https://github.com/emberjs/ember.js/issues/15327 in a comment, thank you!